ansible/verify_config.py

```diff
@@ -0,0 +1,768 @@
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_proxy_is_using(),
+            reason="Proxy setting is not correct ",
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_can_connect_to_docker_registry(),
+            reason="Cannot establish a connection to docker registry. Check your proxy setting, docker proxy setting"
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_can_connect_to_apt_repository(),
+            reason="Cannot establish a connection to apt repository. Check your proxy setting."
+        )
+
+    def _check_if_can_connect_to_apt_repository(self):
+        return_code = Utility.execute_shell("apt-get --simulate upgrade")
+        return return_code == 0
+
+    def _check_if_bridge_is_up(self):
+        return_code = Utility.execute_shell("ifconfig br1")
+        return return_code == 0
+
+    def _check_if_can_connect_to_docker_registry(self):
+        is_docker_installed = Utility.execute_shell("command docker -v", is_shell=True) == 0
+
+        if not is_docker_installed:
+            self.assertion.add_error_details("Docker is not installed on the system. Please install docker")
+            return False
+
+        docker_image_url = f"{Config.DOCKER_REGISTRY_URL}/{Config.DOCKER_IMAGE_NAME}:latest"
+
+        # We need to check for pull here since only pull use the setting from docker-proxy. Unless in the future
+        # there are other alternatives. If fail exit code will be `1` otherwise will be `124` exit code for `timeout`
+        can_pull_image = Utility.execute_shell(f"timeout 1s docker pull {docker_image_url}") == 124
+
+        if not can_pull_image:
+            self.assertion.add_error_details("Not able to connect to docker-registry. Please configure your proxy")
+            return False
+
+        return True
+
+    def _check_if_proxy_is_using(self):
+        group_vars_env = Utility.parse_yml(Config.GROUP_VARS_ENV_FILE)
+
+        env_vars = ["http_proxy", "https_proxy"]
+
+        for var in env_vars:
+            if var not in os.environ:
+                continue
+
+            if var not in group_vars_env or var not in group_vars_env['proxy_env']:
+                self.assertion.add_error_details(
+                    f"'{Formatting.red(var)}' is detected in environment variables "
+                    f"but not in '{Config.GROUP_VARS_ENV_FILE}'",
+                )
+                return False
+
+            if group_vars_env[var] != os.environ[var]:
+                self.assertion.add_error_details(
+                    f"environment '{Formatting.red(var)}' is not the same "
+                    f"as declared in '{Config.GROUP_VARS_ENV_FILE}': "
+                    f"{os.environ[var]} != {group_vars_env[var]}",
+                )
+                return False
+
+        return True
+
+
+class TestBedValidator(Validator):
```

Comment:
What's the relationship between this `TestBedValidator` class and the one with the same name above? Should we rename one of them to avoid confusion and potential issues?

Reply:
I've changed it to `DUTValidator` since it verifies a specific DUT device. Please let me know if it's still confusing.

Reply:
How about `TestbedConnectionValidator`? I think `DUTValidator` is still a bit overlapping with `TestbedValidator`.

Reply:
Yup that's a good one. Let me change to that.